scw/request_test.go

```diff
@@ -83,3 +85,39 @@ func TestGetHeadersWithBody(t *testing.T) {
 
 	testhelpers.Equals(t, expectedHeaders, allHeaders)
 }
+
+func TestSetBody(t *testing.T) {
```

Comment:
Wouldn't it make sense to also test `SetBody` with a file? Because afaik this is not yet covered.